### ISSUE TITLE
How does regularisation affect the training speed? I am getting X100 boost.

### ISSUE THREAD

**Issue by IgorHoholko** [tensorflow/recommenders#646]
Hello!

I am experimenting with the base model from tutorials.
I have noted that after adding `tf.keras.regularizers.l2(0.05) )`  to embeddings layer two things happen:
1.  Metrics start growing - understandable.
2. Speed of trining becomes X50-100 times faster. **Why can it happen?**

One more thing I noted is that when I add a few Dense layers without any regularizers or add `tf.keras.regularizers.l2(1)` with big weight to embedding I am getting all accuracies **Top1, Top3, ..., Top100 equal to 1**, and loss is low and stable. What is happening in this case?

```python
vocab = tf.keras.layers.experimental.preprocessing.StringLookup(vocabulary=vocabulary[slot_name])
embedding = tf.keras.layers.Embedding(vocab.vocabulary_size(), embedding_dim, embeddings_regularizer=tf.keras.regularizers.l2(0.05) )
inputs.append( tf.keras.Sequential([vocab, 
                        embedding, 
                        # tf.keras.layers.Dense(32, activation="relu"),
                        # tf.keras.layers.Dense(32, activation="relu"),
                        ]) )
```

<img width="617" alt="image" src="https://user-images.githubusercontent.com/27057946/221435931-0be8207c-adec-431c-9756-acf6725f1e6e.png">

**Comment by IgorHoholko** [comment]
Thank you @patrickorlando .

Sorry for being not clear. I mean time per step in training loop.

**Comment by patrickorlando** [comment]
Then that is quite odd. Are you using multiple GPUs @IgorHoholko?

**Comment by IgorHoholko** [comment]
I use one RTX 4090.
